### PR TITLE
Preps for v2 Part2 seaice

### PR DESCRIPTION
**Pull request by chengzhuzhang**
Using ncremap  -P mpasseaice for sgs mode. A new routine to slice time-series and conduct per time-slice regridding is required for applying ncremap's sgs mode for mpas seaice.

### CODE REVIEW

File: e3sm_to_cmip/cmor_handlers/sitemptop.py
Comment (tomvothecoder) [inline e3sm_to_cmip/cmor_handlers/sitemptop.py]: This comma was causing a syntax error, which was breaking the GH Actions test suite job. It is removed and the build should be working now.

The rest of the changes are from the code formatting tools.

Comment (chengzhuzhang) [reply e3sm_to_cmip/cmor_handlers/sitemptop.py]: Thank you @tomvothecoder the comma was introduced right before pushing the code without testing again. My bad. Thank you for helping out.

Comment (chengzhuzhang) [reply e3sm_to_cmip/cmor_handlers/sitemptop.py]: Thank you @tomvothecoder the comma was introduced right before pushing the code without testing again. My bad. Thank you for helping out.

Comment (tomvothecoder) [reply e3sm_to_cmip/cmor_handlers/sitemptop.py]: No problem, I add accidental characters like spaces too from time to time.

File: e3sm_to_cmip/mpas.py
Comment (xylar) [inline e3sm_to_cmip/mpas.py:53]:
```suggestion
    ds_out_all = ds_out_all.drop('timeMonthly_avg_iceAreaCell')
```

**Comment by chengzhuzhang** [comment]
Hi @xylar, I updated this PR to be able to use the nco 5.1.1 alpha12 feature, that is to only rely on nco to add mask to non-sea-ice grids and to apply SGS weights. Could you please review again the `mpas.py` and `sitemptop.py`? I will plan to propagate same change to other sea ice variables if the changes are okay. Thank you.

File: e3sm_to_cmip/mpas.py
Comment (xylar) [reply e3sm_to_cmip/mpas.py:80]: I think there's something weird with the if/elif/if going on here.  Presumably, you want to handle:
* mpasseaice and sgs
* mpasseaice without sgs
* mpaso
It seems to me that relying on "sgs" in the mapping file name is a little too fragile.  It would be better to explicitly provide an argument about whether sgs shoudl be use.  There should be an error if you try to use sgs with mpaso, since that isn't supported.  There could be a sanity check and and error or a warning if sgs is requested but the mapping file doesn't contain "sgs" in its name or if sgs is *not* requested and sgs *is* in the name.  But I'll leave this up to you.  What you currently plan is going to work fine as long as the mapping files are named as expected.

**Review by xylar — CHANGES_REQUESTED** [review]
I think the remapping code needs a little clean up of the if/elif/if sequence.

File: e3sm_to_cmip/mpas.py
Comment (chengzhuzhang) [inline e3sm_to_cmip/mpas.py:94]: address @xylar 's review comment regarding to if logic.

**Review by chengzhuzhang — COMMENTED** [review]
@xylar I have addressed your review comment and applied the handler changes to rest of the 8 sea-ice handlers (other than sitemptop). 
@xylar and @czender In my test, all 9 sea-ice variables are generated successfully. Based on my initial assessment, the simass and sithick variables look weird.  Both fields seem not be continuous and with large values occur in a few spots. The data is at `/p/user_pub/e3sm/zhang40/tests/e2c_sea-ice/CMIP6/CMIP/E3SM-Project/E3SM-2-0/historical/r1i1p1f1/SImon` on ACME1. Please let me know if you'd like me to move the files to another location. Thank you!

**Comment by czender** [comment]
I looked at simass. The max value is ~69000 kg/m2. This implies a sea-ice thickness of ~70m.  This is extremely large though not inconceivable along a coastal margin. @xylar will need to weigh-in on whether MPAS-SI is known to produce such values.

**Comment by czender** [comment]
Also, maps with "sgs" in the name were only for the v1 regridder. Using them with recent NCO risks double counting SGS affects, non-conservation, and cats and dogs loving eachother. I think calling NCO with maps with "sgs" in the name should be completely eliminated. There is no need for that anymore.

**Comment by chengzhuzhang** [comment]
Just to add that, the newly generated data was from a E3SM v2 historical simulation for year 1850. Out of curiosity, I also checked the E3SM v1 historical simulation over year 1850-1854, the max of simass (5600 km/m2) and sithick (6 m) are 10 times smaller.

**Comment by chengzhuzhang** [comment]
> Also, maps with "sgs" in the name were only for the v1 regridder. Using them with recent NCO risks double counting SGS affects, non-conservation, and cats and dogs loving eachother. I think calling NCO with maps with "sgs" in the name should be completely eliminated. There is no need for that anymore.

Yes, for the updated code, it will throw an error if the users supply a mapping file with `_sgs_` in the file name.

**Comment by czender** [comment]
I suggest checking the SGS fraction in that v2 gridcell. If SGS is really small then the large mass could be an artifact of dividing be a small number. If SGS is small, try re-running with --rnr_thr=0.01 or some such and see if the overly large maximum becomes a masked gridpoint instead. This is fine-tuning kind of stuff. Seems like we are converging towards completion.

**Comment by chengzhuzhang** [comment]
> I suggest checking the SGS fraction in that v2 gridcell. If SGS is really small then the large mass could be an artifact of dividing be a small number. If SGS is small, try re-running with --rnr_thr=0.01 or some such and see if the overly large maximum becomes a masked gridpoint instead. This is fine-tuning kind of stuff. Seems like we are converging towards completion.

I did try with a 0.05 threshold, but still get the same maximum values for both variables. 
(With rnr_thr = 0.05 it reports 5% more count of missing values than rnr_thr, which suggest the flag is functional.)

**Comment by czender** [comment]
It was worth a try. I'll be curious to hear @xylar's thoughts...

**Comment by chengzhuzhang** [comment]
The identified problematic values seem to occur at one grid point, see below:

Screen Shot 2022-10-21 at 4 17 33 PM

Screen Shot 2022-10-21 at 4 22 02 PM

**Comment by czender** [comment]
If the problematic value is also high in the cells in the input files that contribute to the single point in the destination grid, then this is an MPAS-SI issue (that Adrian/Nicole/Andrew/Elizabeth would probably appreciate hearing about). If not, it is a regridder and/or masking issue that Xylar, me, and you need to deal with...

**Comment by xylar** [comment]
> The data is at /p/user_pub/e3sm/zhang40/tests/e2c_sea-ice/CMIP6/CMIP/E3SM-Project/E3SM-2-0/historical/r1i1p1f1/SImon on ACME1. Please let me know if you'd like me to move the files to another location. Thank you!

I was hoping to take a look today but I have lost access to Chrysalis (my account expired) which also means I lost access to Acme1 because of the requirement for a static IP (I was using Chrysalis as my point to hop over to Acme1).  I'm hoping to have access restored soon but I can't look at the data until then.

**Comment by chengzhuzhang** [comment]
@xylar Not sure if you got your account back. I made a copy of both input and output (cmorized) data to NERSC: /global/cfs/cdirs/e3sm/zhang40/e2c_tests,  so that you and other MPAS sea-ice developers also have access. Thank you.

**Comment by xylar** [comment]
Thanks @chengzhuzhang!  I'll take a look first thing tomorrow.  I'm back in Germany now and it's too late for me to do a good job this evening.

**Comment by chengzhuzhang** [comment]
> Thanks @chengzhuzhang! I'll take a look first thing tomorrow. I'm back in Germany now and it's too late for me to do a good job this evening.

Of course, let me know if you find anything tomorrow. Thank you.

**Comment by xylar** [comment]
@chengzhuzhang (and @czender), I took a careful look at the example MPAS-Seaice output.  Everything looks good from the `e3sm_to_cimp` standpoint.  As @chengzhuzhang noted, there are 2 regions of very thick sea ice, one in Disko Bay in Greenland and one on the western side of the Ross Ice Shelf in Antarctica.  These seem to be related single-cell embayments that should probably have been culled out of the mesh.  It will be important to point this problem out to the sea-ice modelers (if they aren't already aware of it) but it certainly has nothing to do with post processing.

In case we want to have some sea-ice folks check this as well, I put some useful files (vtk versions of the native MPAS-Seaice output that can be viewed in paraview) here on Cori/Perlmutter:
```
/global/cfs/cdirs/e3sm/xylar/e2c_tests/v2.mpassi_input/vtk_files/
```

**Comment by xylar** [comment]
@chengzhuzhang, could this be rebased to make it a little easier to do a final code review?

**Comment by xylar** [comment]
@chengzhuzhang, can you please run a test to make sure this still works as expected following my rebase?  I had to fix several conflicts and I think that happened without a problem but it would be better to catch any errors before this gets merged.

**Review by xylar — APPROVED** [review]
I believe things look good.  The code is not possible to review on GitHub because of the Codecov and formatting commits but I had a careful look at the resulting code and results and I think they make sense.

**Comment by chengzhuzhang** [comment]
@xylar I should have been more thoughtful organizing the commits. Thank you for reviewing the code anyway and for inspect the results. I tested again those 9 sea-ice handlers and they ran successfully. I guess we will hear from sea-ice folks about the spurious points discovered. At this point, for e2c I think we are completing the v2 support journey by merging this PR. This is great team work! Thank you everyone, @xylar @czender @TonyB9000 @tomvothecoder  for making it happen.

**Pull request merged**
